### ISSUE TITLE
ubuntu 22.04 , nginx nginx/1.23.1 

### ISSUE THREAD

**Issue by ashrafalzyoud** [phusion/passenger#2438]
```
The following packages have unmet dependencies:
 libnginx-mod-http-passenger : Depends: nginx-common (< 1.18.1) but 1.23.1-1+ubuntu22.04.1+deb.sury.org+1 is to be installed
E: Unable to correct problems, you have held broken packages.

```

**Comment by CamJN** [comment]
` 1.23.1-1+ubuntu22.04.1+deb.sury.org+1` is not afaict the package available from the official ubuntu repo, the ubuntu repo has `1.18.0-6ubuntu14.1`. If you want to use our packaged Nginx module, you must use your distro's provided Nginx package. If for example you have the repo provided by NGINX setup, you will instead need to [compile a dynamic module compatible with that Nginx.](https://www.phusionpassenger.com/docs/advanced_guides/install_and_upgrade/nginx/install_as_nginx_module.html#dynamic-module)

**Comment by schorsch** [comment]
i do have the same issue with the hard dependencies, because there is a bug in the ubuntu 22 supplied version of nginx 1.18 leading to lots of ssl errors in the nginx log (though the web app seems to run fine) see => 

- https://github.com/openssl/openssl/issues/18866
- https://askubuntu.com/questions/1405100/i-have-problem-with-new-ubuntu-22-04-and-openssl-3-0-2 

i know this not the fault of you guys nor the one of the openssl or nginx suppliers but actually ubuntu not getting an up-to-date nginx package. i've just been through a little dependency hell, also because i am using some nginx-extra packages, which where taken from the ubuntu deps and overwrote the beforehand installed nginx from the nginx dep .. just a mess

sorry to be a bit off-topic here, but i would strongly advice anybody to just use the supplied deps by phusion and the nginx ones from the distro. Self-compilation leaves you without upgrades and deeper links into your system like systemd scripts...YMMV

**Comment by heapxor** [comment]
is there some solution / fix? thx

**Comment by CamJN** [comment]
@heapxor i gave the solution in my reply above, use the Ubuntu repo for Nginx or compile your own passenger module.

**Comment by nulf** [comment]
@heapxor If you wan't to use the latest from ppa:ondrej I fixed a similar issue(on Ubuntu 22.04 with ondrejs ppa installed) with the following:
(answer found on https://askubuntu.com/a/1106505 - all credit goes to George Udosen@SO)

After you have done a `sudo apt purge nginx` you need to force the installation of the package from the new ppa. Not sure if there is a reference left somewhere in the OS even if the package has been purged. Anyways, here we go:
```
sudo apt install -f nginx -y
sudo apt install -f nginx-core -y
sudo apt install -f nginx-common -y
sudo apt install -f nginx-full y
```

Edit: And I had to run them one by one. Don't really know why  ¯\\_(ツ)_/¯
